diff --git a/node_modules/workerize-loader/dist/index.js b/node_modules/workerize-loader/dist/index.js
index 831afaf..5042d1a 100644
--- a/node_modules/workerize-loader/dist/index.js
+++ b/node_modules/workerize-loader/dist/index.js
@@ -73,7 +73,7 @@ loader.pitch = function (request) {
                 var entry = compilation.entries[0].resource;
                 if (current.resource !== entry)
                     { return; }
-                var exports = compilation.__workerizeExports || (compilation.__workerizeExports = {});
+                var exports = CACHE[entry] || (CACHE[entry] = {});
                 if (decl.id) {
                     exports[decl.id.name] = true;
                 } else if (decl.declarations) {
@@ -91,8 +91,9 @@ loader.pitch = function (request) {
             { return cb(err); }
         if (entries[0]) {
             worker.file = entries[0].files[0];
+            var entry = entries[0].entryModule.resource;
             var contents = compilation.assets[worker.file].source();
-            var exports = Object.keys(CACHE[worker.file] = compilation.__workerizeExports || CACHE[worker.file] || {});
+            var exports = Object.keys(CACHE[entry] || {});
             if (options.inline) {
                 worker.url = "URL.createObjectURL(new Blob([" + (JSON.stringify(contents)) + "]))";
             } else {
